### PR TITLE
Log an event for each Service Status change

### PR DESCRIPTION
**Pull request by PipoCanaja**
Hello

Here is a proposal for a pièce of code to log (using log_events) each service status change onto the device itself. 

Let me know if we should create the "getStatusText()" function to convert the int to "OK", "Critical" etc etc . I searched for such a function but did not find it. 

I also add to use a hack for the SQL cause it would prevent the device_id valueto be available in the resulting $service when device_attribs is empty for the device. If you have a better way to suggest, I'll be happy to use it. 

PipoCanaja

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: check-services.php [hunk 1]
```diff
@@ -55,7 +55,9 @@
     }
 }
 
-$sql = 'SELECT * FROM `devices` AS D'
+// need to specify D.device_id as D_device_id because device_id gets overriden in the resulting raw by A.device_id if no attribs are defined for this device
+// TODO: a cleaner way
+$sql = 'SELECT D.device_id as D_device_id,D.*,S.*,A.* FROM `devices` AS D'
```
Comment (laf) [inline check-services.php]: Can be changed to 

```
SELECT D.*,S.*, attrib_value FROM `devices` AS `D`
```

File: includes/services.inc.php [hunk 2]
```diff
@@ -113,7 +113,7 @@ function discover_service($device, $service)
     echo "$service ";
 }
 
-function poll_service($service)
+function poll_service($service, $device)
```
Comment (laf) [inline includes/services.inc.php]: Should be able to remove this second param now.

**Comment by PipoCanaja** [comment]
Changes are done. I added the WIP tag to test that into production in my environnement to see if it is all right with more services than test machine. Should take a few days. 

PipoCanaja

**Comment by PipoCanaja** [comment]
Works well in my production system as well. OK for me.

Bye
PipoCanaja

**Pull request merged**